hibernate-core/src/main/java/org/hibernate/property/access/internal/AccessStrategyHelper.java

```diff
@@ -82,9 +83,56 @@ public static AccessType getAccessType(Class<?> containerJavaType, String proper
 				&& field.isAnnotationPresent( Access.class )
 				&& !field.isAnnotationPresent( Transient.class )
 				&& !Modifier.isStatic( field.getModifiers() ) ) {
-			return AccessType.FIELD;
+			AccessType accessType = getAccessTypeOrNull(field);
+			return accessType == null ? AccessType.FIELD : accessType;
+			//return AccessType.FIELD;
```

Comment:
The `if` condition assures we're always going to have an `@Acess` annotation here

```suggestion
			return field.getAnnotation( Access.class ).value();
```

Reply:
Suggestion leads to a Checker Framework error
`[dereference.of.nullable] dereference of possibly-null reference field.getAnnotation(Access.class)`.

A will change to `return getAccessTypeOrNull( field );`

Reply:
Just use `org.hibernate.processor.util.NullnessUtil#castNonNull(T)` to promote `field.getAnnotation( Access.class )` to a non-nullable value, since we already checked it exists on the field.